hack/update-model.sh

```diff
@@ -1,6 +1,47 @@
 #!/bin/bash -e
 
-# This script ensures that all the api-model submodules used across the project are updated to the latest verison.
+# ==============================================================================
+# Script: update-model.sh
+#
+# This script ensures that all the `ocm-api-model` submodules used across the
+# project are updated to the latest version or pinned to a specific commit SHA.
+#
+# The OCM SDK can be generated simply by running the following after all changes have been made:
+#
+# ./hack/update-model.sh
+# make update
+#
+# USAGE:
+#   ./update-model.sh [COMMIT_SHA]
+#
+# ARGUMENTS:
+#   COMMIT_SHA (optional) - If provided, all listed modules will be pinned to
+#                           this specific commit instead of updating to latest.
+#
+# EXAMPLES:
+#   ./update-model.sh
+#     → Updates all modules to their latest versions.
+#
+#   ./update-model.sh f67fb59980981bdc81d95d1379a82da5bcec57bf
+#     → Pins all modules to the specified commit SHA.
+#
+# ==============================================================================
+
+# Optional commit SHA
+COMMIT_SHA="$1"
+
+# Remote to validate commit SHA against
+REPO_URL="https://github.com/openshift-online/ocm-api-model.git"
+
+if [ -n "$COMMIT_SHA" ]; then
+  echo "Checking if commit SHA exists in remote repo..."
+  if git ls-remote "$REPO_URL" | grep -q "^$COMMIT_SHA"; then
+    echo "✅ Commit $COMMIT_SHA found in remote repository."
+  else
+    echo "❌ Commit SHA $COMMIT_SHA not found in remote $REPO_URL"
+    exit 1
+  fi
```

Comment:
Why not just let `go get ` to it's thing?
e.g getting a non existing `hjdhjsd898` commit 
```
go get github.com/openshift-online/ocm-api-model/model@hjdhjsd898                                                      [
go: github.com/openshift-online/ocm-api-model/model@hjdhjsd898: invalid version: unknown revision hjdhjsd898
```

Reply:
I don't mind how to validate (git log or go get) - I was just looking to make sure if a user gives a wrong SHA it will fail. As long as that happens - I'm happy :)

Reply:
Ok. I am removing this validation than.